command/lock.go

```diff
@@ -165,6 +170,10 @@ func (c *LockCommand) Run(args []string) int {
 	go func() {
 		if err := c.startChild(script, childDone, passStdin); err != nil {
 			c.Ui.Error(fmt.Sprintf("%s", err))
+
+			if c.childExitCode {
+				c.childReturnedError = true
```

Comment:
^ updated that comment if you are reading in email, it's much simpler to poll. Let me know if this makes sense - I can help work up a patch based on your branch.